### DISCUSSION TITLE
Connection to a webrtc stream over Wireguard

### DISCUSSION THREAD

**Discussion by eprouty** [bluenviron/mediamtx#3924] [Questions]
### Question

Hello, I have mediamtx running on a Pi5 with the default configuration and a `cam` section that works perfectly on my local network.

When connecting to `serverip:8889/cam` through wireguard VPN, running in an LXC on my Proxmox server, the webrtc connection always fails after the "checking" phase.  When looking at the network debug tools all of the http calls are succeeding, so the tunnel is connecting to the instance fine.

The logs show the following:
```
Oct 30 19:57:15 mainsail mediamtx[4808]: 2024/10/30 19:57:15 INF [WebRTC] [session 5222d778] created by <wireguardip>:61614
Oct 30 19:57:17 mainsail mediamtx[4808]: 2024/10/30 19:57:17 INF [WebRTC] [session 3041c207] created by <wireguardip>:61620
Oct 30 19:57:25 mainsail mediamtx[4808]: 2024/10/30 19:57:25 INF [WebRTC] [session 5222d778] closed: deadline exceeded while waiting connection
Oct 30 19:57:27 mainsail mediamtx[4808]: 2024/10/30 19:57:27 INF [WebRTC] [session 3041c207] closed: deadline exceeded while waiting connection
```

To add to this, I know the tunnel is configured correctly because I can switch to using TCP for webrtc it works... although at significantly worse quality, and with sporadic disconnects, so I'm hoping to get UDP working over the VPN.

Any help is appreciated, thank you!

Server version: 1.9.3
Replication:
1. Setup default mediamtx.yml with a `cam` section
2. Setup wireguard VPN and connect a client through it
3. Try to connect to the webrtc endpoint.

**Comment by adzmailru** [comment]
First you need to see the wireguard settings. With the correct settings, all your ip addresses should ping each other.

**Comment by aler9** [comment]
You have to make sure that `webrtcAdditionalHosts` contains the server host, and that the port contained inside `webrtcLocalUDPAddress` can be accessed by clients, through the VPN. That's all.

The fact that all HTTP calls are succeeding is not an indicator of the fact that the server has been set up properly, since in WebRTC the handshake happens on a different channel (HTTP, `webrtcAddress`) with respect to data (either UDP / `webrtcLocalUDPAddress` or TCP / `webrtcLocalTCPAddress`).

In particular, there's a guide about that in the README:
https://github.com/bluenviron/mediamtx?tab=readme-ov-file#solving-webrtc-connectivity-issues
